src/main/java/org/junit/internal/runners/statements/FailOnTimeout.java

```diff
@@ -103,6 +107,20 @@ public Builder withLookingForStuckThread(boolean enable) {
         }
 
         /**
+         * Specifies the description of this test. If this is provided, the
+         * thread generated when evaluating this statement will be given a
+         * name corresponding with the description.
+         *
+         * @param description a description of the test, to be used when
+         *                    naming the generated thread
+         * @return {@code this} for method chaining.
+         */
+        public Builder withDescription(Description description) {
+            this.description = description;
```

Comment:
I think it would be simpler to call `description.getDisplayName()` here, and replace the "description" field with a "threadName" field of type `String` with a default value of "Time-limited test"